base/shell/explorer/traywnd.cpp

```diff
@@ -2991,12 +3040,30 @@ class CTrayWindow :
             ::EnableMenuItem(hMenu, ID_SHELL_CMD_CASCADE_WND, MF_BYCOMMAND | MF_ENABLED);
             ::EnableMenuItem(hMenu, ID_SHELL_CMD_TILE_WND_H, MF_BYCOMMAND | MF_ENABLED);
             ::EnableMenuItem(hMenu, ID_SHELL_CMD_TILE_WND_V, MF_BYCOMMAND | MF_ENABLED);
+            if (m_Arrangement != NONE)
+            {
+                CStringW strCaption((m_Arrangement == TILED) ? MAKEINTRESOURCEW(IDS_TRAYWND_UNDO_TILE)
+                                                             : MAKEINTRESOURCEW(IDS_TRAYWND_UNDO_CASCADE));
+                MENUITEMINFOW mii = { sizeof(mii) };
+                GetMenuItemInfoW(hMenu, ID_SHELL_CMD_UNDO_ACTION, FALSE, &mii);
+                mii.fMask = MIIM_ID | MIIM_TYPE;
+                mii.wID = ID_SHELL_CMD_UNDO_ACTION;
```

Comment:
Why do you update the id?

Reply:
Good point, unecessary here. Fixed

Reply:
Has the comment been implemented? or?

Reply:
Yes but not pushed yet. Sorry my mistake, I will do it asap.